### ISSUE TITLE
Rebuilds improvement

### ISSUE THREAD

**Issue by rapgro** [displaylink-rpm/displaylink-rpm#125]
Can you use Copr for rebuilds? It supports at least Fedora and CentOS (EPEL) as well as Mageia.
See also #76 .

https://docs.pagure.org/copr.copr/user_documentation.html#what-i-can-build-in-copr
https://developer.fedoraproject.org/deployment/copr/copr-cli.html

**Comment by rapgro** [comment]
OpenBuildService is an alternative.

https://openbuildservice.org/help/manuals/obs-user-guide/par.working.html

**Comment by pvalena** [comment]
I've investigated COPR (would've done it myself) few weeks ago, but I found out you need to have a opensource license for that (not shipping any proprietary code)... basically same as in Fedora.

https://docs.pagure.org/copr.copr/user_documentation.html#what-i-can-build-in-copr

**Comment by elguero** [comment]
Thanks for investigating and confirming.  I had a feeling that would be the case.

**Comment by rapgro** [comment]
Why not evdi at least and a script to additionally download the proprietary part? That's allowed IMHO. I am going to find some time to create and suggest a spec file, maybe also for package review.

**Comment by pvalena** [comment]
> Why not evdi at least and a script to additionally download the proprietary part? That's allowed IMHO. I am going to find some time to work on the spec file, maybe also for package review.

Yes, you're right, it could be done in a similar way (not sure about "during dnf install"), but don't you have to compile it also? It would require user to have preinstalled GCC/llvm toolchain, and all the devel libs... I'm not familiar with any restrictions about this in Fedora repos, but in any case, RPMFusion may be the right way to go.

> Besides there are some community packages built in OpenBuildService for openSUSE.

Maybe you could copy / compare the spec files, in that case?

**Comment by rapgro** [comment]
@pvalena It's about DisplayLinkManager as the central component and officially shipped only as binary blob for Ubuntu, that's what a script should downlaod and properly install after dnf install. The kernel is tainted with evdi module, built from dkms having always trouble with major changes to kernel API as seen recently, see discussion in other issues here. But I can agree, rpmfusion seems the better repository to use therefore, besides dkms should be replaced with akmod.

**Comment by elguero** [comment]
Adding reference to PR: https://github.com/displaylink-rpm/displaylink-rpm/pull/97
